src/Paginator.php

```diff
@@ -661,7 +661,7 @@ public function getIterator()
         try {
             return $this->getCurrentItems();
         } catch (Throwable $e) {
-            throw new Exception\RuntimeException('Error producing an iterator', null, $e);
+            throw new Exception\RuntimeException('Error producing an iterator', (int) $e->getCode(), $e);
```

Comment:
Type cast is not needed since `getCode()` has return type is `int`

Reply:
Psalm complains if there is no type cast.